plugins/org.eclipse.n4js/src/org/eclipse/n4js/n4idl/scoping/NonVersionAwareContextScope.java

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright (c) 2018 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.n4idl.scoping;
+
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.n4js.n4idl.N4IDLGlobals;
+import org.eclipse.n4js.n4idl.versioning.VersionUtils;
+import org.eclipse.n4js.scoping.utils.AbstractDescriptionWithError;
+import org.eclipse.n4js.validation.IssueCodes;
+import org.eclipse.n4js.validation.JavaScriptVariantHelper;
+import org.eclipse.n4js.validation.ValidatorMessageHelper;
+import org.eclipse.n4js.xtext.scoping.FilteringScope;
+import org.eclipse.n4js.xtext.scoping.IEObjectDescriptionWithError;
+import org.eclipse.xtext.resource.IEObjectDescription;
+import org.eclipse.xtext.scoping.IScope;
+
+import com.google.inject.Inject;
+
+/**
+ * A {@link NonVersionAwareContextScope} which wraps all parent elements that are considered to be {@code @VersionAware}
+ * (cf. {@link VersionUtils#isVersionAwareContext(EObject)}), with an issue indicating, that the current context does
+ * not allow references to elements in version-aware contexts.
+ *
+ * This scope wraps filtered description either with
+ * {@link IssueCodes#IDL_ELEMENT_CAN_ONLY_BE_REFERENCED_FROM_VA_CONTEXT} if the reference is made from a type-version
+ * supporting language variant or with {@link IssueCodes#IDL_INVALID_REFERENCE_TO_VERSION_AWARE_ELEMENT} otherwise.
+ *
+ * For instantiation please see {@link NonVersionAwareContextScope.Provider}.
+ */
+public class NonVersionAwareContextScope extends FilteringScope {
+
+	/**
+	 * Injectable provider for {@link NonVersionAwareContextScope}.
+	 */
+	public static final class Provider {
```

Comment:
In all our scope classes (except one, see below) we do not use such a factory class but instead simply pass the injected services and other required arguments (context, etc.) to the constructor and let clients directly invoke the constructor (for example, see constructor of `VisibilityAwareCtorScope`). For simplicity and consistency, please use that approach here as well.

If this is not possible for some reason, at least align this code to the one case where we use such a factory class, i.e. `MemberScopeFactory` in `MemberScope`. This means:
1. rename `Provider` to `NonVersionAwareContextScopeFactory`,
2. rename method `Provider#get()` to `create`.